extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js

```diff
@@ -1552,4 +1556,122 @@ describes.sandboxed('amp-ad-network-doubleclick-impl', {}, () => {
+      forceExperimentBranch(impl.win, CORRELATOR_CLEAR_EXP_NAME,
+          CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT);
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(impl.win.ampAdPageCorrelator).to.not.be.ok;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL))
+          .to.be.false;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT))
+          .to.be.true;
+    });
+
+    it('does not clear if in control', () => {
+      forceExperimentBranch(impl.win, CORRELATOR_CLEAR_EXP_NAME,
+          CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL);
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL))
+          .to.be.true;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT))
+          .to.be.false;
+    });
+
+    it('does not clear if in neither branch', () => {
+      forceExperimentBranch(impl.win, CORRELATOR_CLEAR_EXP_NAME, null);
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(impl.win.ampAdPageCorrelator).to.equal(12345);
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL))
+          .to.be.false;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT))
+          .to.be.false;
+    });
+
+    it('attempts to select into branch', () => {
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(
+          impl.win.experimentBranches[CORRELATOR_CLEAR_EXP_NAME] !== undefined);
+    });
+
+    it('does not attempt to select into branch if SRA', () => {
+      impl.useSra = true;
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(impl.win.ampAdPageCorrelator).to.equal(12345);
+      expect(
+          impl.win.experimentBranches[CORRELATOR_CLEAR_EXP_NAME] === undefined);
+    });
+
+    it('does not attempt to select into branch if no correlator', () => {
+      impl.win.ampAdPageCorrelator = undefined;
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(
+          impl.win.experimentBranches[CORRELATOR_CLEAR_EXP_NAME] === undefined);
+    });
+
+    it('does not attempt to select into branch if not pause', () => {
+      visabilityState = VisibilityState.VISIBLE;
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(
+          impl.win.experimentBranches[CORRELATOR_CLEAR_EXP_NAME] === undefined);
+    });
+
+    it('only registers one on viewability change listener', () => {
```

Comment:
what is this testing? is this supposed to be 'only register one onVisibilityChangedHandler' ?